[BugFix] Sticky plugin max-width calculation is insufficient

JerBear90 opened the pull request:
The calculation between `newElemWidth` and `pdng` seems to be insufficient.
The max-width should be calculated by `newElemWidth - (pdng * 2)`

This PR addresses the following Issue:
Sticky plugin makes div larger #8427
https://github.com/zurb/foundation-sites/issues/8427

I have provided a test at test/visual/sticky/adjust-max-width

JerBear90 committed:
[BugFix] Sticky plugin max-width calculation is insufficient

The calculation between newElemWidth and pdng seems to be insufficient.
The max-width should be calculated by newElemWidth - (pdng * 2)
Addresses Issue with
Sticky plugin makes div larger #8427

JerBear90 committed:
Testing [Bug Fix] Sticky max-width calculations

This example recalculates the max-width based off of the container. The
padding is multiplied by 2 and subtracted from the element to give the
max-width.

JerBear90 committed:
Bump to version

kball commented:
Lets try 2 things to make sure this is a general fix:

1. Remove the implementation of sticky within the test page; it should get your new version automatically by including the dist/foundation.js

2. Build a testcase with padding-left and padding-right being different to make sure the doubling works generally, or if we should instead be explicitly pulling out padding-left and padding-right and adding them.

JerBear90 committed:
Merge remote-tracking branch 'zurb/develop' into develop

JerBear90 committed:
Address Sticky - Reinit

To Address the issues: I've taken a look at the Sticky plugin and I believe that the window.one is restricting the re-initialization of the plugin once destroyed. line 1116 before: $(window).one('load.zf.sticky', function () { after: $(window).trigger('load.zf.sticky', function () { I've replaced the .one() with .trigger() and it seems to respond as expected. The height of the sticky element regains it height after being destroyed. I haven't received any errors in the console, yet. I would greatly appreciate any feedback about using this method.

ncoden commented:
Hi @Jflem1290. Can you rebase your work ?

poke @kball

kball commented:
There's a lot of conflicts in here and I'm no longer sure what was being fixed in this.  Based on the title though, I suspect this is addressed in https://github.com/zurb/foundation-sites/pull/9298

Going to close for now.  @Jflem1290 please feel free to reopen with a rebased update if you think there is something being lost not addressed in #9298
